src/controller/slack.go

```diff
@@ -71,3 +72,60 @@ func (c *Controller) SaveAuthDetails(ctx context.Context, authDetails model.Slac
 
 	return nil
 }
+
+// GetStickerSearchResult shows up the search result
+func (c *Controller) GetStickerSearchResult(ctx context.Context, channelID, teamID, userID, text string) error {
+	slackService := c.getSlackService(ctx, teamID)
+
+	result, totalCount, err := c.cloudinary.SearchByTag(ctx, text)
+	if err != nil {
+		c.logger.Err(err).Msg("cloudinary.SearchByTag failed")
+		return err
+	}
+
+	if totalCount < 1 {
+		return fmt.Errorf("not found")
+	}
+
+	imageURL := result[0].URL
+
+	return slackService.ShowStickerPreview(ctx, userID, channelID, text, imageURL)
+}
+
+// CancelSticker to close sticker preview block
+func (c *Controller) CancelSticker(ctx context.Context, teamID, channelID, responseURL string) error {
+	slackService := c.getSlackService(ctx, teamID)
+	return slackService.CancelStickerPreview(ctx, channelID, responseURL)
+}
+
+// SendSticker to send sticker
+func (c *Controller) SendSticker(ctx context.Context, teamID, userID, channelID, responseURL string, sticker model.StickerBlockActionValue) error {
+	c.logger.Info().
+		Str("channelID", channelID).
+		Msg("sending sticker")
+	slackService := c.getSlackService(ctx, teamID)
+	return slackService.SendStickerToChannel(ctx, userID, channelID, responseURL, sticker)
+}
+
+// ShuffleSticker to shuffle sticker
+func (c *Controller) ShuffleSticker(ctx context.Context, teamID, userID, channelID, responseURL string, sticker model.StickerBlockActionValue) error {
+	slackService := c.getSlackService(ctx, teamID)
+	result, totalCount, err := c.cloudinary.SearchByTag(ctx, sticker.Tag)
+	if err != nil {
+		c.logger.Err(err).Msg("cloudinary.SearchByTag failed")
+		return err
+	}
+
+	if sticker.Index >= totalCount {
+		sticker.Index = 0
+	}
+
+	if len(result) > 0 {
+		sticker.ImgURL = result[sticker.Index].URL
+		if err := slackService.ShuffleStickerPreview(ctx, userID, channelID, responseURL, sticker); err != nil {
+			return err
+		}
```

Comment:
add some logs here.

Reply:
I also cannot see the code that's choosing the random sticker.

Reply:
check here -> https://github.com/noornee/slack-stickers-be/blob/f5ca5948cac370819a829fa8c8dd39fafd83bd95/src/slack/utils.go#L90